src/imp/privsep.c

```diff
@@ -164,28 +164,31 @@ privsep_t * privsep_init (privsep_child_f fn, void *arg)
     return (ps);
 }
 
-int privsep_destroy (privsep_t *ps)
+int privsep_wait (privsep_t *ps)
 {
     int status = 0;
-
-    if (ps->wfd > 0)
-        close (ps->wfd);
-    if (ps->rfd > 0)
-        close (ps->rfd);
-
     if (privsep_is_parent (ps)) {
         if (ps->cpid > (pid_t) 0) {
-            int status;
             kill (SIGTERM, ps->cpid);
             if (waitpid (ps->cpid, &status, 0) < 0)
                 status = -1;
+            ps->cpid = (pid_t) -1;
         }
     }
-
-    free (ps);
     return (status == 0 ? 0 : -1);
 }
 
+void privsep_destroy (privsep_t *ps)
+{
+    if (ps) {
+        if (ps->wfd > 0)
+            close (ps->wfd);
+        if (ps->rfd > 0)
+            close (ps->rfd);
```

Comment:
I realize this is not code that originated in this PR, but since 0 is a valid file descriptor, and these values are initialized from `upfds` and `ppfds` which use -1 to designate a closed fd, should this be checking for >= 0?

Reply:
Yes, I think you're correct. I'll add a fix to this PR.